### DISCUSSION TITLE
Error loading shared libraries

### DISCUSSION THREAD

**Discussion by schembriaiden** [abraunegg/onedrive#2761] [Q&A] [answered]
Hiya,

I have your onedrive package installed on my Arch Linux install. It used to work fine but all of a sudden it stopped working with the following error:

```
❯ onedrive
onedrive: error while loading shared libraries: libphobos2-ldc-shared.so.108: cannot open shared object file: No such file or directory
```

Would someone be able to help?

**Comment by abraunegg** [comment, marked as answer]
@vikingnope 
This is not an issue with this client, but your chosen distribution and using the AUR package.

Please read https://aur.archlinux.org/packages/onedrive-abraunegg for details on how to resolve this issue

**Comment by cyb3rko** [comment]
It seems to be an issue again.  
The comments in the AUR package do not work for me, does anybody have a solution?

**Reply by the-horo** [reply]
Can you post any build/logs and the actual output when running the program. Building and running https://aur.archlinux.org/packages/onedrive-abraunegg worked fine in a docker container.

**Reply by cyb3rko** [reply]
The build works fine, I'm getting the following error when running `onedrive --sync`:

`onedrive: error while loading shared libraries: libphobos2-ldc-shared.so.110: cannot open shared object file: No such file or directory`

**Reply by the-horo** [reply]
That doesn't really answer my question. What commands are you using to build `onedrive`? Did you try updating your system first (mostly the `ldc` package)?

**Reply by cyb3rko** [reply]
Simply `yay onedrive`.  
And I have tried using dmd or llvm19-libs, both didn't work either.

Maybe I should switch to a Docker container

**Reply by the-horo** [reply]
1. This package is named `onedrive-abraunegg` on the AUR. You should be running `yay onedrive-abraunegg`, not `yay onedrive` to install it.
2. The error is showing that the `onedrive` you are running has been built with `ldc-1.40`. The current version in the arch linux repositories is [1.41.0](https://archlinux.org/packages/extra/x86_64/ldc/). Please update your system

If it's not working even after that can you try following the [manual](https://wiki.archlinux.org/title/Arch_User_Repository) build guide for aur packages. Go through steps 1, 2, and, 3 of the *Installing and upgrading packages* section, but, before you install the package, run the `onedrive` executable from inside the build directory. My guess is that the onedrive in your `PATH` is not the one that is being installed by `yay`

**Reply by the-horo** [reply]
> Maybe I should switch to a Docker container

If you don't want to troubleshoot this, sure

**Reply by cyb3rko** [reply]
> 1. This package is named `onedrive-abraunegg` on the AUR. You should be running `yay onedrive-abraunegg`, not `yay onedrive` to install it.

`yay onedrive-abraunegg` searches for matching packages, I then select the right one.

> 2. The error is showing that the onedrive you are running has been built with ldc-1.40. The current version in the arch linux repositories is [1.41.0](https://archlinux.org/packages/extra/x86_64/ldc/). Please update your system

I update my system at least once a day, everything's up-to-date.

> My guess is that the onedrive in your PATH is not the one that is being installed by yay

Good idea, I will check that

**Reply by abraunegg** [reply]
@cyb3rko 
> Simply `yay onedrive`.

If you did this you will have the 'skilion' client installed somewhere as the same binary name. Please read https://github.com/abraunegg/onedrive?tab=readme-ov-file#project-background for details.

As to why there is 'onedrive' and 'onedrive-abraunegg' on AUR .. it is because the package maintainer of the 'onedrive' AUR package has an issue with how and why I started this fork and the reasoning I provided.

Potentially complain on AUR to have the now defunct package removed and replaced so that there is no name ambiguity.

**Reply by cyb3rko** [reply]
I don't think so, because I selected package 5 and later package 11, which both seem to be the correct AURs (`aur/onedrive-abraunegg` and `aur/onedrive-abraunegg-git`)

![grafik](https://github.com/user-attachments/assets/48f9d45e-4621-4db3-9aaa-61e5a6442261)

**Comment by AidanofVT** [comment]
I resolved this by simply reinstalling `onedrive-abraunegg`. I opted for the "rebuild all" option offered by yay, but I don't know whether that actually contributed to it working. I also admit that it's possible that something else solved the problem in the recently, and I'm just noticing it now.
